### PR TITLE
Allow vector integrands

### PR DESCRIPTION
**Pull request by swertz**
Take advantage of Cuba's feature allowing vector integrands: each component is integrated in parallel, the grid being optimised for all of them (obviously, this only works if all components are similar).

Does not change much for the user: `MoMEMta::computeWeights()` is not modified. To define the components, simply give several input tags as arguments to the `integrand()` function in Lua:
```
integrand("integrand1::sum", "integrand2::sum")
```
integrates the output of modules `integrand1` and `integrand2` separately.

### CODE REVIEW

File: core/src/MoMEMta.cc [hunk 1]
```diff
@@ -137,11 +143,19 @@ std::vector<std::pair<double, double>> MoMEMta::computeWeights(const std::vector
     // Output from cuba
     long long int neval = 0;
     int nfail = 0;
-    double mcResult = 0, prob = 0, error = 0;
-
+    double* mcResult = new double[m_n_components]; 
+    double* prob = new double[m_n_components]; 
+    double* error = new double[m_n_components];
```
Comment (blinkseb) [inline core/src/MoMEMta.cc]: You have a leak here for the 3 arrays (no `delete[]` associated). Please use a `std::unique_ptr<double[]>` here

**Comment by swertz** [comment]
Of course... Better like that?

**Comment by swertz** [comment]
Darn...
```
Saving to: ‘ROOT-6.06.02_Python-2.7_gcc-4.9.3_Ubuntu_14.04_x86_64.tar.xz’

15% [=====>                                 ] 25,873,149  18.5KB/s  eta 99m 17s
```

**Comment by blinkseb** [comment]
Can you possibly add a unit test for the `lua::set_final_module` function (in `tests/lua.cc`)?

**Comment by swertz** [comment]
Done!

**Comment by blinkseb** [comment]
Awesome thanks a lot!

**Pull request merged**